### PR TITLE
Make useIdiomaticEndpoints default to true

### LINKED ISSUE #93: The argument methodNameStyle of service macro should default to "Unchanged"
By default, when the "muSrcGenIdiomaticEndpoints" setting is true, the generated service has "methodNameStyle = Capitalize", this is a surprise to most grpc developers.

In addition, if muSrcGenIdiomaticEndpoints = false, the generated service does not have namespace set. Thus, there is not option available that allows us to have namespace and unchanged method names.

When we use protobuf as a rpc protocol, we often use some protobuf tools as well, like grpcui, or BloomRPC to test grpc services. However, due to the fact that grpc services generated by mu-srcgen is not compatible to its proto file, those tools do not work anymore.

In conclusion, we expect the generated grpc service to be like the following, to comply with conventions of grpc ecosystem.
```scala
@service(Protobuf, Identity, namespace = Some("example.seed.protocol.proto"), methodNameStyle = Unchanged) trait PeopleService[F[_]]
```

### PR DESCRIPTION
**Pull request by L-Lavigne**
Fixes #93 via https://github.com/higherkindness/skeuomorph/pull/342 and the resulting Skeuomorph release 0.0.26.

Also cleans up the Model by removing unused classes, and updates the tests and documentation.

### CODE REVIEW

File: core/src/main/scala/higherkindness/mu/rpc/srcgen/avro/AvroSrcGenerator.scala [hunk 1]
```diff
@@ -103,6 +103,7 @@ final case class AvroSrcGenerator(
       })
       .map(generateFrom(_, serializationType))
 
+  //TODO: implement this using higherkindness.skeuomorph.mu.Protocol.fromAvroProtocol, as is done in ProtoSrcGenerator
```
Comment (juanpedromoreno) [inline core/src/main/scala/higherkindness/mu/rpc/srcgen/avro/AvroSrcGenerator.scala:106]: This is probably part of https://github.com/higherkindness/sbt-mu-srcgen/pull/101, right?

File: plugin/src/main/scala/higherkindness/mu/rpc/srcgen/SrcGenPlugin.scala [hunk 2]
```diff
@@ -155,7 +155,7 @@ object SrcGenPlugin extends AutoPlugin {
       }
     },
     muSrcGenCompressionType := NoCompressionGen,
-    muSrcGenIdiomaticEndpoints := false,
+    muSrcGenIdiomaticEndpoints := true,
```
Comment (juanpedromoreno) [inline plugin/src/main/scala/higherkindness/mu/rpc/srcgen/SrcGenPlugin.scala:158]: We'll probably need to update [this section](https://higherkindness.io/mu-scala/reference/source-generation#advanced-settings) in the mu-scala docs 👍

**Review by juanpedromoreno — APPROVED** [review]
Thanks @L-Lavigne !

File: core/src/main/scala/higherkindness/mu/rpc/srcgen/avro/AvroSrcGenerator.scala (hunk 1 above)
Comment (L-Lavigne) [reply core/src/main/scala/higherkindness/mu/rpc/srcgen/avro/AvroSrcGenerator.scala:106]: Oh, I was just making a note of the discrepancy and forgot there was an issue to address this. :-)

File: plugin/src/main/scala/higherkindness/mu/rpc/srcgen/SrcGenPlugin.scala (hunk 2 above)
Comment (L-Lavigne) [reply plugin/src/main/scala/higherkindness/mu/rpc/srcgen/SrcGenPlugin.scala:158]: Indeed, I've got a documentation PR for my-scala coming up.

**Review by dzanot — APPROVED** [review]

File: core/src/main/scala/higherkindness/mu/rpc/srcgen/avro/AvroSrcGenerator.scala (hunk 1 above)
Comment (dzanot) [reply core/src/main/scala/higherkindness/mu/rpc/srcgen/avro/AvroSrcGenerator.scala:106]: I'm rebasing #101 off of this in anticipation of yours being merged first :D

**Pull request merged**